backend/api-gateway/clients/inventory_client.go

```
@@ -0,0 +1,284 @@
+	inventoryAddr := fmt.Sprintf("%s:%s", cfg.Services.Inventory.Host, cfg.Services.Inventory.Port)
+	logger.Info("Connecting to inventory service", zap.String("address", inventoryAddr))
+
+	// Set up connection with retry logic
+	var conn *grpc.ClientConn
+	var err error
+	maxRetries := 3
+	retryInterval := time.Second * 2
+
+	for i := 0; i < maxRetries; i++ {
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+		conn, err = grpc.DialContext(
```

golangci-lint (1.64.8)

SA1019: grpc.DialContext is deprecated: use NewClient instead.  Will be supported throughout 1.x. 

(staticcheck)